runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -77,6 +67,7 @@ static ExecutionEnvironment createBatchExecutionEnvironment(
 
     // depending on the master, create the right environment.
     if ("[local]".equals(flinkMasterHostPort)) {
+      flinkConfiguration.setString("taskmanager.memory.managed.size", "2048m");
```

Comment:
I have the feeling this won't be reliable enough. Why not instead `taskmanager.memory.managed.fraction`?

Reply:
It will set the `taskmanager.memory.managed.size` as [128MB](https://github.com/apache/flink/blob/b90d1a269661f4e5ca132213a03169fc9011aaee/flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorResourceUtils.java#L139) for MiniCluster if it's not set. I think set `taskmanager.memory.managed.fraction"` doesn't take effect here. Thoughts? :)

Reply:
I'm hesitant with this default because it will always pre-allocate 2GB of memory which won't be used most of the time, except for the one large record test case you mentioned.

We could set I'd go for something like https://github.com/apache/flink/blob/42a56f4c75693773e21fa2dea45df640c2d7f9da/flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorProcessUtils.java#L287 based on the memory available.

Actually, that is what the Flink 1.8 code used to do: https://github.com/apache/flink/blob/60d9b96456f142f8d18d5882016840a00159403e/flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java#L296

So let's just check the free memory and use a fraction for memory managed memory by default. What do you think?

Reply:
Thanks @mxm, Sounds good to me ;)

Reply:
Cool, thanks for the changes.